tests/test_vision_data.py

```diff
@@ -0,0 +1,19 @@
+import pytest
+from fastai import *
+from fastai.vision import *
+
+@pytest.fixture(scope="module")
+def path(request):
+    path = untar_data(URLs.MNIST_TINY)
+    d = defaults.device
+    defaults.device = torch.device('cpu')
+    def _final(): defaults.device = d
+    request.addfinalizer(_final)
+    return path
+
+def test_from_name_re(path):
+    fnames = get_files(path/'train', recurse=True)
+    pat = r'\/([^/]+)\/\d+.png$'
+    data = ImageDataBunch.from_name_re(path, fnames, pat, ds_tfms=(rand_pad(2, 28), []))
+    assert data.c == 2
+    assert all(["3" in data.classes, "7" in data.classes]) is True
```

Comment:
Could you add something to test the elements in the training and validation set are correctly labeled?